### PR TITLE
Issue 506 | Gateway: fix permissions for save results

### LINKED ISSUE #506: Result method should be protected
### Steps to reproduce the problem

`result` method in `gateway` has public access and should be protected:

https://github.com/Qiskit-Extensions/quantum-serverless/blob/main/gateway/api/views.py#L162

### PR DESCRIPTION
**Pull request by IceKhan13**
### Summary

Gateway: fix permissions for save results + update on JWT access fields

### Details and comments

Closes #506

### CODE REVIEW

**Review by pacomf — APPROVED** [review]

File: client/quantum_serverless/core/provider.py [hunk 1]
```diff
@@ -584,5 +584,5 @@ def _fetch_token(self, username: str, password: str):
         if not gateway_response.ok:
             raise QuantumServerlessException(gateway_response.text)
 
-        gateway_token = json.loads(gateway_response.text).get("access_token")
+        gateway_token = json.loads(gateway_response.text).get("access")
```
Comment (Tansito) [inline client/quantum_serverless/core/provider.py:587]: Oh, when did the `access_token` keyword change to `access`?

Comment (psschwei) [reply client/quantum_serverless/core/provider.py:587]: One of the dependencies released a new major version which got pulled in for the latest builds. @IceKhan13 suggested we may want to pin the deps in #509 , +1 for me.

Comment (IceKhan13) [reply client/quantum_serverless/core/provider.py:587]: Basically this comment https://github.com/Qiskit-Extensions/quantum-serverless/pull/509#issuecomment-1538686258

Comment (Tansito) [reply client/quantum_serverless/core/provider.py:587]: Yes! I couldn't answer before due to the problems in GitHub haha. I skipped that comment, thank you team 😄

**Pull request merged**